File: 735-asteroid-collision/asteroid-collision.java
import java.util.Stack;

class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> st = new Stack<>();
        int count = 0;

        for (int i = 0; i < asteroids.length; i++) {
            if (asteroids[i] > 0) {
                st.push(asteroids[i]);
                count++;
            } else {
                while (count > 0 && Math.abs(asteroids[i]) > st.peek() && st.peek() > 0) {
                    st.pop();
                    count--;
                }
                if (count == 0 || st.peek() < 0) {
                    st.push(asteroids[i]);
                    count++;
                } else if (Math.abs(asteroids[i]) == st.peek()) {
                    st.pop();
                    count--;
                }
            }
        }

        int[] result = new int[count];
        for (int i = count - 1; i >= 0; i--) {
            result[i] = st.pop();
        }

        return result;
    }
}

File: 735-asteroid-collision/asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        st = []
        for i in asteroids:
            st.append(i)
            if st and i < 0:
                while len(st) > 1:
                    ele1 = st.pop()
                    ele2 = st[-1]
                    if ele2 > 0:
                        if ele2 < abs(ele1):
                            st.pop()
                            st.append(ele1)
                        elif ele2 == abs(ele1):
                            st.pop()
                            break
                        else:
                            break
                    else:
                        st.append(ele1)
                        break
        return st